### PR TITLE
intersight_rest_api module and integration tests.

### PR DESCRIPTION
**Pull request by dsoper2**
Fix intersight module_utils issues when using POST/PATCH/DELETE.

##### SUMMARY

New module to configure resources using the Cisco Intersight REST API.

##### ISSUE TYPE

- New Module Pull Request

##### COMPONENT NAME

intersight_rest_api
##### ADDITIONAL INFORMATION

```paste below

```

### CODE REVIEW

**Comment by movinalot** [comment]
shipit

**Review by movinalot — APPROVED** [review]
shipit

File: lib/ansible/module_utils/remote_management/intersight.py [hunk 1]
```diff
@@ -186,11 +186,14 @@ def call_api(self, **options):
...
             if not re.match(r'2..', str(info['status'])):
-                raise RuntimeError(info['status'], info['msg'])
+                raise RuntimeError(info['status'], info['msg'], info['body'])
         except Exception as e:
             self.module.fail_json(msg="API error: %s " % str(e))
 
-        return json.loads(response.read())
+        json_response = {}
+        if response.length > 0:
+            json_response = json.loads(response.read())
```
Comment (dagwieers) [inline lib/ansible/module_utils/remote_management/intersight.py]: I would simply do:
```python
        if response.length > 0:
            return json.loads(response.read())
        return {}
```

**Review by dagwieers — APPROVED** [review]
LGTM, just one remark

**Pull request merged**